test/Tests/src/Data/Text_Spec.enso

```diff
@@ -337,6 +339,24 @@ spec =
             kshi_chars = [2325, 2381, 2359, 2367]
             Text.from_utf_16 kshi_chars . should_equal kshi
 
+        Test.specify "should insert text at a positive index position" <|
+            hello_world.insert 0 cruel . should_equal " CruelHello World!"
+            hello_world.insert 5 cruel . should_equal "Hello Cruel World!"
+            hello_world.insert (hello_world.length - 1) cruel . should_equal "Hello World Cruel!"
+            hello_world.insert hello_world.length cruel . should_equal "Hello World! Cruel"
+
+        Test.specify "should report Index_Out_Of_Bounds_Error when inserting text at an invalid positive index position" <|
+            hello_world.insert (hello_world.length + 1) cruel . should_fail_with Index_Out_Of_Bounds_Error
+
+        Test.specify "should insert text at a negative index position" <|
+            hello_world.insert -1 cruel . should_equal "Hello World! Cruel"
+            hello_world.insert -5 cruel . should_equal "Hello Wo Cruelrld!"
+            hello_world.insert -(hello_world.length) cruel . should_equal "H Cruelello World!"
+            hello_world.insert -(hello_world.length + 1) cruel . should_equal "" CruelHello World!"
+
+        Test.specify "should report Index_Out_Of_Bounds_Error when inserting text at an invalid negative index position" <|
+            hello_world.insert -(hello_world.length + 2) cruel . should_fail_with Index_Out_Of_Bounds_Error
```

Comment:
I think you should definitely add tests which check that insertion respects grapheme cluster boundaries instead of working on raw codepoints.

You can use the constants `kshi` and `facepalm` defined above. May also use combining accents, like `'a\u{301}'`, as seen in other tests.